src/autoconf/main.py

```
@@ -38,19 +44,20 @@ def exit_handler(signum, frame):
-        logger.info("Swarm mode detected")
-        controller = SwarmController(docker_host)
-    elif kubernetes:
-        logger.info("Kubernetes mode detected")
-        controller = IngressController()
-    else:
-        logger.info("Docker mode detected")
-        controller = DockerController(docker_host)
 
-    # Wait for instances
+def run_on_kubernetes_ha_mode(controller):
+    config.load_incluster_config()
```

Ruff (0.8.2)

Local variable `config` referenced before assignment

(F823)